lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -77,32 +80,113 @@ def __init__(
         self._grpc_client_manager = grpc_client_manager
 
     async def _run(self) -> None:
-        subscribers = await self._get_subscribers()
-        if subscribers is not None:
-            self._process_subscribers(subscribers)
+        in_sync = await self._check_subscribers_in_sync()
+        if in_sync:
+            return
+
+        resync = await self._sync_subscribers()
+        if not resync:
+            return
+
+        ret = await self._get_all_subscribers()
+        if ret is None:
+            return
 
-    async def _get_subscribers(self) -> [SubscriberData]:
+        subscribers, flat_digest = ret
+        self._update_flat_digest(flat_digest)
+        self._process_subscribers(subscribers)
+
+    async def _check_subscribers_in_sync(self) -> bool:
+        """
+        Check if the local subscriber data is up-to-date with the cloud by
+        comparing flat digests
+
+        Returns:
+            boolean value for whether the local data is in sync
+        """
+        subscriberdb_cloud_client = self._grpc_client_manager.get_client()
+        req = CheckSubscribersInSyncRequest(
+            flat_digest=Digest(
+                md5_base64_digest=self._store.get_current_digest(),
+            ),
+        )
+        try:
+            res = await grpc_async_wrapper(
+                subscriberdb_cloud_client.CheckSubscribersInSync.future(
+                    req,
+                    self.SUBSCRIBERDB_REQUEST_TIMEOUT,
+                ),
+                self._loop,
+            )
+        except grpc.RpcError as err:
+            logging.error(
+                "Check subscribers in sync request error! [%s] %s", err.code(),
+                err.details(),
+            )
+            return False
+        return res.in_sync
+
+    async def _sync_subscribers(self) -> bool:
```

Comment:
Yeah let's just remove this fn for now (you can save it in a txt file to use during milestone 3)